lib/checkPullRequestLabels.js

```diff
@@ -14,6 +15,93 @@ var matchChangelogLabelWithRegex = function(changelogLabel) {
   return regExpForChangelogLabel.test(changelogLabel);
 };
 
+/**
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ * @param {String} changelogLabel
+ */
+const canAssignProjectOwner = (pullRequest, changelogLabel) => {
+  const isExcluded = EXCLUDED_LABELS.some((label) =>
+    changelogLabel.startsWith(label)
+  );
+  if (isExcluded) {
+    return true;
+  }
+  const projectOwner = getProjectOwner(changelogLabel);
+  const reviewers = pullRequest.requested_reviewers.map(
+    reviewer => reviewer.login
+  );
+  const canReview = reviewers.includes(projectOwner);
+
+  return canReview;
+};
+
+/**
+ * @param {String} changelogLabel
+ */
+const getProjectOwner = (changelogLabel) => {
+  const labelSubstrings = changelogLabel.split('@');
+  const projectOwner = labelSubstrings[labelSubstrings.length - 1].trim();
+  return projectOwner
+};
+
+
+/**
+ * Assign pull request to reviewers.
+ * @param {import('probot').Context} context
+ */
+const assignAllCodeowners = async (context) => {
+  // Github automatically requests for review from codeowners.
+  /**
+   * @type {import('probot').Octokit.PullsGetResponse} context
+   */
+  const pullRequest = context.payload.pull_request;
+  const assignees = pullRequest.requested_reviewers.map(
+    (reviewer) => reviewer.login
+  );
+  if (assignees.length > 0) {
+    // Assign Reviewers.
+    const assignmentParams = context.repo({
+      issue_number: pullRequest.number,
+      assignees,
+    });
+    await context.github.issues.addAssignees(assignmentParams);
+
+    // Ping codeowners.
+    const message =
+      'Assigning @' +
+      assignees.join(', @') +
+      ' for the first pass review of this PR. Thanks!';
+
+    const commentParams = context.repo({
+      issue_number: pullRequest.number,
+      body: message,
+    });
+    await context.github.issues.createComment(commentParams);
+  } else {
+    // Oppiabot can't assign here since the only codeowner is the PR author,
+    // hence we ping PR author to assign appropriate reviewer as they will have
+    // access to assign the required reviewer.
+    const author = pullRequest.user.login;
+    const assignmentParams = context.repo({
+      issue_number: pullRequest.number,
+      assignees: [author],
+    });
+    await context.github.issues.addAssignees(assignmentParams);
+
+    // Ping author.
+    const message =
+      'Hi @' +
+      author +
+      ' please assign the required reviewer(s) for this PR. Thanks!';
+
+    const commentParams = context.repo({
+      issue_number: pullRequest.number,
+      body: message,
+    });
+    await context.github.issues.createComment(commentParams);
+  }
+};
+
 /**
  * @param {import('probot').Context} context
  */
```

Comment:
Add a detailed docstring for the function describing the exact flow of how we assign the reviewer.

Reply:
Also, let us add docstrings for all functions in this file, if missing.

Reply:
Done.

Reply:
Add a list of bullets as to how we assign - the exact flow:
1. Check for codeowners
2. ...

Reply:
Done.